spdx/package.go

```diff
@@ -15,22 +15,29 @@ type Supplier struct {
 	SupplierType string
 }
 
-// UnmarshalJSON takes a supplier in the typical one-line format and parses it into a Supplier struct.
-// This function is also used when unmarshalling YAML
-func (s *Supplier) UnmarshalJSON(data []byte) error {
-	// the value is just a string presented as a slice of bytes
-	supplierStr := string(data)
-	supplierStr = strings.Trim(supplierStr, "\"")
+// Validate verifies that all the required fields are present.
+// Returns an error if the object is invalid.
+func (s Supplier) Validate() error {
+	// SupplierType is allowed to be empty if Supplier is "NOASSERTION"
+	if s.Supplier == "" || (s.SupplierType == "" && s.Supplier != "NOASSERTION") {
+		return fmt.Errorf("invalid Supplier, missing fields. %+v", s)
+	}
 
-	if supplierStr == "NOASSERTION" {
-		s.Supplier = supplierStr
+	return nil
+}
+
+// FromString parses a string into a Supplier.
+// These stings take the form: "<SupplierType>: <Supplier>"
+func (s *Supplier) FromString(value string) error {
+	if value == "NOASSERTION" {
+		s.Supplier = value
 		return nil
 	}
 
-	supplierFields := strings.SplitN(supplierStr, ": ", 2)
+	supplierFields := strings.SplitN(value, ": ", 2)
```

Comment:
Here and for other similar instances elsewhere in the PR, same question as above about whether it's useful to leverage the `extractSubs()` function from the tag-value loader. If not, that's fine, but I'm guessing it may need to handle the possibility of more than 1 whitespace character after the colon.

Reply:
This PR now uses the code from `extractSubs` instead of my implementation that only supported one space.